lib/rake/build.rake

```diff
@@ -63,8 +63,9 @@ namespace :build do
         ChatClient.log 'Migrating <b>dashboard</b> database...'
         RakeUtils.rake 'db:setup_or_migrate'
 
-        # Update the schema cache file, except for production which always uses the cache.
-        unless rack_env?(:production)
+        # Update the schema cache file, except for production which always uses the cache,
+        # and on adhoc where updating schema cache can cause merge conflicts.
+        unless [:production, :adhoc].include?(rack_env)
```

Comment:
Only `staging` needs to create `schema_cache.yml` because the staging system's Rails database is the source of truth for dashboard's schema. Maybe also systems operating with the `test` environment type should also generate `schema_cache.yml` to verify it's the same as what was committed to the repository?